pilot/util/tracereport.py

```diff
@@ -0,0 +1,117 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Authors:
+# - Alexey Anisenkov, alexey.anisenkov@cern.ch, 2017
+# - Pavlo Svirin, pavlo.svirin@cern.ch, 2018
+
+import time
+
+import hashlib
+#import commands
+from pilot.util.container import execute
+
+import socket
+
+import pilot.util.https
+from sys import exc_info
+
+import logging
+logger = logging.getLogger(__name__)
+
+
+class TraceReport(dict):
+
+    def __init__(self, *args, **kwargs):
+
+        defs = {
+            'eventType': '',           # sitemover
+            'eventVersion': 'pilot2',  # pilot version
+            'protocol': None,          # set by specific sitemover
+            'clientState': 'INIT_REPORT',
+            'localSite': '',           # localsite
+            'remoteSite': '',          # equals remotesite (pilot does not do remote copy?)
+            'timeStart': None,         # time to start
+            'catStart': None,
+            'relativeStart': None,
+            'transferStart': None,
+            'validateStart': None,
+            'timeEnd': None,
+            'dataset': '',
+            'version': None,
+            'duid': None,
+            'filename': None,
+            'guid': None,
+            'filesize': None,
+            'usr': None,
+            'appid': None,
+            'hostname': '',
+            'ip': '',
+            'suspicious': '0',
+            'usrdn': '',
+            'url': None,
+            'stateReason': None,
+            'uuid': None,
+            'taskid': ''
+        }
+
+        super(TraceReport, self).__init__(defs)
+        self.update(dict(*args, **kwargs))  # apply extra input
+
+    # sitename, dsname, eventType
+    def init(self, job):
+
+        data = {
+            'clientState': 'INIT_REPORT',
+            'usr': hashlib.md5(job.prodUserID).hexdigest(),  # anonymise user and pilot id's
+            'appid': job.jobid,
+            'usrdn': job.prodUserID,
```

Comment:
Currently the DN is not read from the job description; I have added it in my branch - will try to create a pull request later today. It will be called 'produserid', i.e. lower case only, so change this to job.produserid.